AmplifyPlugins/API/AWSAPICategoryPlugin/Configuration/AWSAPICategoryPluginConfiguration.swift

```diff
@@ -34,15 +60,82 @@ public struct AWSAPICategoryPluginConfiguration {
         let endpoints = try AWSAPICategoryPluginConfiguration.endpointsFromConfig(config: config,
                                                                                   apiAuthProviderFactory: apiAuthProviderFactory,
                                                                                   authService: authService)
-        self.init(endpoints: endpoints)
+        let interceptors = try AWSAPICategoryPluginConfiguration.interceptorsForEndpoints(endpoints,
+                                                     apiAuthProviderFactory: apiAuthProviderFactory,
+                                                     authService: authService)
+
+        self.init(endpoints: endpoints,
+                  interceptors: interceptors,
+                  apiAuthProviderFactory: apiAuthProviderFactory,
+                  authService: authService)
+
+    }
+
+    /// Registers an interceptor for the provided API endpoint
+    /// - Parameter interceptor: operation interceptor used to decorate API requests
+    /// - Parameter toEndpoint: API endpoint name
+    mutating func addInterceptor(_ interceptor: URLRequestInterceptor,
+                                 toEndpoint apiName: APIEndpointName) {
+        interceptors[apiName]?.addInterceptor(interceptor)
+    }
+
+    /// Returns all the interceptors registered for `apiName` API endpoint
+    /// - Parameter apiName: API endpoint name
+    /// - Returns: request interceptors
+    internal func interceptorsForEndpoint(named apiName: APIEndpointName?) -> [URLRequestInterceptor] {
```

Comment:
i think apiName should always be required since resolving the `EndpointConfig` is done upstream (getting first GraphQL/first REST/specific EndpointConfig, etc)